clang-tools-extra/test/clang-tidy/checkers/bugprone/unchecked-optional-access.cpp

```diff
@@ -180,3 +180,44 @@ void std_forward_rvalue_ref_safe(absl::optional<int>&& opt) {
 
   std::forward<absl::optional<int>>(opt).value();
 }
+
+namespace std {
+
+template<typename T>
+class optional {
+public:
+	template <typename U>
+  optional& operator=(const U &u){
+		data = u;
+	}
+private:
+  T data;
+};
+
+template <typename T>
+class vector {
+public:
+  vector() {}
+  T &operator[](unsigned long index) { return data[index]; }
+  bool empty() { return size == 0UL; }
+
+private:
+  T *data;
+  unsigned long size;
+};
+
+} // namespace std
+
+struct LinearGradient {
+  struct Stop {
+    std::optional<float> position;
+  };
+  std::vector<Stop> stops;
```

Comment:
Is it necessary for the repro to have this surrounding struct of `LinearGradient`? Will it crash with just `std::vector<Stop>`?